### PR TITLE
Coverage for: factory/consensus

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage
  
## Proposed changes
- added unittests for factory/consensus

## Testing procedure
- standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by oxsyai — APPROVED** [review]

**Review by oxsyai — COMMENTED** [review]
Wrong github account :)

**Review by SebastianMarian — APPROVED** [review]

File: integrationTests/mock/cryptoComponentsStub.go [hunk 1]
```diff
@@ -14,11 +14,13 @@ import (
 // CryptoComponentsStub -
 type CryptoComponentsStub struct {
 	PubKey                  crypto.PublicKey
+	PublicKeyCalled         func() crypto.PublicKey
 	PrivKey                 crypto.PrivateKey
 	P2pPubKey               crypto.PublicKey
 	P2pPrivKey              crypto.PrivateKey
 	PubKeyBytes             []byte
 	PubKeyString            string
+	PrivKeyBytes            []byte
```
Comment (iulianpascalau) [inline integrationTests/mock/cryptoComponentsStub.go:23]: is this still necessary?

File: testscommon/rounderMock.go [hunk 2]
```diff
@@ -1,12 +1,14 @@
 package testscommon
```
Comment (iulianpascalau) [inline testscommon/rounderMock.go:1]: please change the filename to roundHandlerMock.go

File: integrationTests/mock/cryptoComponentsStub.go (hunk 1 above)
Comment (sstanculeanu) [reply integrationTests/mock/cryptoComponentsStub.go:23]: removed it

File: factory/consensus/consensusComponents.go [hunk 3]
```diff
@@ -710,33 +668,63 @@ func (ccf *consensusComponentsFactory) addCloserInstances(closers ...update.Clos
...
+	if check.IfNil(args.StatusComponents) {
+		return errors.ErrNilStatusComponentsHolder
+	}
+	if check.IfNil(args.StatusComponents.OutportHandler()) {
+		return errors.ErrNilOutportHandler
+	}
+	if check.IfNil(args.ScheduledProcessor) {
+		return errors.ErrNilScheduledProcessor
+	}
+	if check.IfNil(args.StatusCoreComponents) {
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/consensus/consensusComponents.go:729]: not used inside factory, only forwarded

File: factory/consensus/consensusComponents.go [hunk 4]
```diff
@@ -710,33 +668,63 @@ func (ccf *consensusComponentsFactory) addCloserInstances(closers ...update.Clos
...
-	if check.IfNil(blockchain) {
-		return errors.ErrNilBlockChainHandler
+func checkArgs(args ConsensusComponentsFactoryArgs) error {
+	if check.IfNil(args.CoreComponents) {
+		return errors.ErrNilCoreComponentsHolder
 	}
-	marshalizer := ccf.coreComponents.InternalMarshalizer()
-	if check.IfNil(marshalizer) {
-		return errors.ErrNilMarshalizer
+	if check.IfNil(args.CoreComponents.GenesisNodesSetup()) {
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/consensus/consensusComponents.go:675]: not used inside factory, only forwarded

**Review by gabi-vuls — APPROVED** [review]
Systemtest passed. 
@@                            Log scanner                                  @@
##                            more_factory_tests_p4                        ##
================================================================================
- Known Warnings              10
- New Warnings                1
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================
- block hash does not match                  12037
- wrong nonce in block                       4041
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    0
================================================================================
-  No jailed nodes on the testnet
================================================================================

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**